server/src/main/java/org/elasticsearch/action/admin/indices/settings/get/GetSettingsResponse.java

```diff
@@ -45,8 +66,17 @@ public GetSettingsResponse(ImmutableOpenMap<String, Settings> indexToSettings) {
 
     public String getSetting(String index, String setting) {
```

Comment:
would you mind adding some javadoc and explaining that we look first among the settings, then among the default ones which are there only when includeDefaults was true in the request?